### PR TITLE
Initial backstage anc buildkite boilerplate

### PR DESCRIPTION
**Pull request by alexsapran**
This commit adds the initial boilerplate for initializing the backstage and buildkite resources.

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Declare a Backstage Component that represents your application.
+---
+# yaml-language-server: $schema=https://json.schemastore.org/catalog-info.json
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: elastic-agent-system-metrics
```
Comment (jlind23) [inline catalog-info.yaml:7]:
```suggestion
  name: elastic-agent-changelog-tool
```

**Review by jlind23 — CHANGES_REQUESTED** [review]

File: catalog-info.yaml (hunk 1 above)
Comment (mrodm) [inline catalog-info.yaml:7]:
```suggestion
  name: elastic-agent-changelog-tool
  description: Tooling to manage the changelog for beats, Elastic Agent and Fleet Server
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  owner: group:ingest-fp
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: elastic-agent-changelog-tool
+      description: Buildkite pipeline for the elastic-agent-changelog-tool project
+    spec:
+      repository: elastic/elastic-agent-changelog-tool
```
Comment (mrodm) [inline catalog-info.yaml:37]:
```suggestion
      branch_configuration: "main v0.*"
      repository: elastic/elastic-agent-changelog-tool
```

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      pipeline_file: ".buildkite/pipeline.yml"
+      provider_settings:
+        build_pull_request_forks: false
+        build_pull_requests: true # requires filter_enabled and filter_condition settings as below when used with buildkite-pr-bot
+        build_tags: true
+        filter_enabled: true
+        filter_condition: |
```
Comment (mrodm) [inline catalog-info.yaml:44]:
```suggestion
        filter_condition: >-
```

**Review by mrodm — CHANGES_REQUESTED** [review]

**Comment by endorama** [comment]
Thanks for the contribution! Let me know if you need help in moving this forward (it seems not) but I would love to see it included in Backstage so big :+1: from me!

File: .buildkite/pull-requests.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "jobs": [
+      {
+         "enabled": true,
+         "pipelineSlug": "elastic-agent-changelog-tool",
+         "allow_org_users": true,
+         "allowed_repo_permissions": ["admin", "write"],
+         "allowed_list": [ ],
+         "set_commit_status": false,
+         "build_on_commit": false,
```
Comment (mrodm) [inline .buildkite/pull-requests.json:10]: As it is the main pipeline, I would set here these two fields as true

```suggestion
         "set_commit_status": true,
         "build_on_commit": true,
```

**Review by mrodm — APPROVED** [review]
:+1:

**Pull request merged**